sxt/execution/device/for_each.cc

```diff
@@ -44,4 +43,11 @@ concurrent_for_each(basit::index_range rng,
     co_await std::move(fut);
   }
 }
+
+xena::future<>
+concurrent_for_each(basit::index_range rng,
```

Comment:
Should this have its own doc comment?

Reply:
Added some documentation to the header file.